openedx/openedx-platform
# fix: downgrade django-storages to 1.14.3

## Description

A recent upgrade to `django-storages` from `1.14.3` to `1.14.4` broke the ability to import a course. The course was able to upload but after uploading the file was not able to be found when the import moved to the "Unpacking" step. This change impacts the "Developer" and "Authors".

This bug was introduced in #35125.

## Supporting information

>the import is broken in edx-platform's master branch by the most recent version of django-storages (1.14.4). I'm not sure what is the proper fix, but pinning it to 1.14.3 in master should work.
The culprit are [these two lines](https://github.com/jschneier/django-storages/blob/b37d53e1209f6333de71502a9cb34d8cffdc7e9f/storages/backends/s3.py#L582-L583) added in `1.14.4` that make [file_exists_in_storage](https://github.com/openedx/edx-platform/blob/475b49c3caad559a8cf85c84fffa676de42d899f/cms/djangoapps/contentstore/tasks.py#L528-L538) function always return `False`, as [the default value of ](https://github.com/jschneier/django-storages/blob/b37d53e1209f6333de71502a9cb34d8cffdc7e9f/storages/backends/s3.py#L409) `AWS_S3_FILE_OVERWRITE` is `True`.

## Testing instructions

1. Open a course in Studio
2. Navigate to the import page
3. Add a file
4. Confirm that file successfully imports and course outline is updated
